test/test_datasets.py

```diff
@@ -41,106 +41,6 @@
-                self.generic_classification_dataset_test(dataset, num_images=len(data["imgs"]))
-
-    def test_places365_transforms(self):
-        expected_image = "image"
-        expected_target = "target"
-
-        def transform(image):
-            return expected_image
-
-        def target_transform(target):
-            return expected_target
-
-        with places365_root() as places365:
-            root, data = places365
-
-            dataset = torchvision.datasets.Places365(
-                root, transform=transform, target_transform=target_transform, download=True
-            )
-            actual_image, actual_target = dataset[0]
-
-            self.assertEqual(actual_image, expected_image)
-            self.assertEqual(actual_target, expected_target)
-
-    def test_places365_devkit_download(self):
-        for split in ("train-standard", "train-challenge", "val"):
-            with self.subTest(split=split):
-                with places365_root(split=split) as places365:
-                    root, data = places365
-
-                    dataset = torchvision.datasets.Places365(root, split=split, download=True)
-
-                    with self.subTest("classes"):
-                        self.assertSequenceEqual(dataset.classes, data["classes"])
-
-                    with self.subTest("class_to_idx"):
-                        self.assertDictEqual(dataset.class_to_idx, data["class_to_idx"])
-
-                    with self.subTest("imgs"):
-                        self.assertSequenceEqual(dataset.imgs, data["imgs"])
-
-    def test_places365_devkit_no_download(self):
-        for split in ("train-standard", "train-challenge", "val"):
-            with self.subTest(split=split):
-                with places365_root(split=split) as places365:
-                    root, data = places365
-
-                    with self.assertRaises(RuntimeError):
-                        torchvision.datasets.Places365(root, split=split, download=False)
-
-    def test_places365_images_download(self):
-        for split, small in itertools.product(("train-standard", "train-challenge", "val"), (False, True)):
-            with self.subTest(split=split, small=small):
-                with places365_root(split=split, small=small) as places365:
-                    root, data = places365
-
-                    dataset = torchvision.datasets.Places365(root, split=split, small=small, download=True)
-
-                    assert all(os.path.exists(item[0]) for item in dataset.imgs)
-
-    def test_places365_images_download_preexisting(self):
-        split = "train-standard"
-        small = False
-        images_dir = "data_large_standard"
-
-        with places365_root(split=split, small=small) as places365:
-            root, data = places365
-            os.mkdir(os.path.join(root, images_dir))
-
-            with self.assertRaises(RuntimeError):
-                torchvision.datasets.Places365(root, split=split, small=small, download=True)
```

Comment:
This test is missing. It basically makes sure we don't re-download anything (and therefore delete everything downloaded previously) if there is already a data folder present. The new test should also `assertRaises(RuntimeError)` when trying to `create_dataset(download=True)`.

Reply:
If I do create_dataset(download=True) twice, wouldn't it create two temporary folders and hence not throw any error?

Reply:
Not sure if I understand. If you use `create_dataset(download=True)` the fake data will be injected and that should be picked up by the dataset instantiation. So basically:

```python
def test_images_download_preexisting(self):
     with self.assertRaises(RuntimeError):
        with self.create_dataset(download=True):
            pass
```

Why would you need it to be created twice?

Reply:
Makes sense, thanks!